File: 0501-0600/0506.relative-ranks.cpp
// 坑, 有可能小于两个人
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& nums) {
        if (nums.empty()) return {};
        vector<int> inds(nums.size());
        iota(inds.begin(), inds.end(), 0);
        sort(inds.begin(), inds.end(), [&](const int a, const int b) {
            return nums[a] > nums[b];
        });
        vector<string> ans(nums.size());
        ans[inds[0]] = "Gold Medal";
        if (nums.size() >= 2)
            ans[inds[1]] = "Silver Medal";
        if (nums.size() >= 3)
            ans[inds[2]] = "Bronze Medal";
        for (int i = 3; i < nums.size(); ++i) {
            ans[inds[i]] = to_string(i + 1);
        }
        return ans;
    }
};

File: 0501-0600/0506-relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        gs = ["Gold Medal","Silver Medal","Bronze Medal"]
        def name(i):
            if i < 3:
                return gs[i]
            return str(i + 1)
        idx = list(range(len(score)))
        idx.sort(key=lambda i: score[i], reverse=True)
        ans = [None for _ in range(len(score))]
        for j, i in enumerate(idx):
            ans[i] = name(j)
        return ans
